extension_helpers/_setup_helpers.py

```diff
@@ -85,6 +111,24 @@ def get_extensions(srcdir='.'):
                         [os.path.join(main_package_dir, '_compiler.c')])
         ext_modules.append(ext)
 
+    use_limited_api = False
+    abi = get_distutils_option('py_limited_api', ['bdist_wheel'])
+    if abi:
+        version_info = _abi_to_version_info(abi)
+        if version_info:
+            use_limited_api = True
+
+    if use_limited_api:
+        log.info(
+            'Targeting PEP 384 limited API supporting Python >= %d.%d',
+            *version_info)
+        version_hex = _version_info_to_version_hex(*version_info)
+        for ext in ext_modules:
+            ext.py_limited_api = True
+            ext.define_macros.append(('Py_LIMITED_API', version_hex))
+    else:
+        log.warn(_PEP_384_WARNING)
```

Comment:
I don't understand the limitations of the limited API really, but does numpy compile against it? I am just wondering if this warning is too keen, i.e. if most of our users (scientific python packages with numpy C dep) are likely to be able to make use of this?

Reply:
Downstream packages that compile against and link with Numpy can use the limited API. I maintain one such package. Here are the wheels: https://pypi.org/project/ligo.skymap/#files

Reply:
The major limitation is that Cython support for the limited API is very, very new: cython/cython#3223

Reply:
I wonder if you could make astropy core compile with this set :thinking: that would certainly be a good indicator that most things could use it.

Thanks for the extra info.

Reply:
So just to make sure I understand, people will see the warning if they don't opt-in to this? If so I wonder if we should perhaps not do that, or at least make it clear to people how to get rid of the warning if they don't want to opt in to the limited API?